test/WebJobs.Script.Tests.Integration/Host/StandbyManagerTests.cs

```diff
@@ -112,6 +112,7 @@ public async Task StandbyMode_EndToEnd_LinuxContainer()
 
             var vars = new Dictionary<string, string>
             {
+                { EnvironmentSettingNames.AzureWebsitePlaceholderMode, "1" },
```

Comment:
This shoudn't be needed, right? What changed here?

Reply:
Relying on the host to set this up itself was problematic. Since we cache the InStandbyMode value if it's false, we had a race that I was hitting and we'd never enter placeholder mode. We need this set up before the host starts.

In a real scenario, this will be auto-set b/c it gets set in the Main() method. But that method doesn't run here as we're using a TestServer.

Reply:
Ah - you had to add this because you moved that startup code it seems? The test previously was verify that that code ran - not it won't be.

Reply:
The startup code was previously sitting in Startup.cs (see here: https://github.com/Azure/azure-functions-host/blob/6aa4d9f82b5b3afe6ece8f76606cf80a64466a0f/src/WebJobs.Script.WebHost/Startup.cs#L20-L24). 

But that introduced a subtle race -- if anyone checked IsStandbyMode before that code ran, it would cache it as 'false' and never change it (as this is meant to be one-way). It's possible to check this value in a Logger before this happens as ASP.NET can create loggers really early in the pipeline. So this code needs to happen ASAP -- likely before we ever start the host. And ideally before we even build the host.

I'd love to have some kind of "Initializer" (not a factory) that we can run on types before we use them for something like this, but that doesn't exist.

@fabiocav and I chatted about this yesterday and we thought this would be okay since we expect something else to set this value for us before the host starts. We expect Antares to do that for us in Windows, and we expect our Main() to do this in Linux.

Reply:
Ok. We probably just need a unit test then on the startup code/method that verifies that it puts the host in standby.

Reply:
Yeah, good point. I can pull that together as well. We want to make sure no one removes that code.